apps/src/code-studio/components/header/ProjectRemix.jsx

```diff
@@ -0,0 +1,62 @@
+/* globals dashboard, appOptions */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import {refreshProjectName} from '../../headerRedux';
+
+class ProjectRemix extends React.Component {
+  static propTypes = {
+    isSignedIn: PropTypes.bool,
+    lightStyle: PropTypes.bool,
+    refreshProjectName: PropTypes.func.isRequired
+  };
+
+  remixProject = () => {
+    if (
+      dashboard.project.getCurrentId() &&
+      dashboard.project.canServerSideRemix()
+    ) {
+      dashboard.project.serverSideRemix();
+    } else if (!this.props.isSignedIn) {
+      window.location.assign(
+        `/users/sign_in?user_return_to=${window.location.pathname}`
+      );
```

Comment:
This was true of the original code too, but this seems like a good time to ask: Why don't we have to urlencode the `user_return_to` parameter here?

Reply:
Is there a built-in guarantee that `window.location.pathname` is already urlencoded? It seems reasonable that there would be